pkg/repositories/transformers/signal.go

```diff
@@ -0,0 +1,134 @@
+package transformers
+
+import (
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/models"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+
+	"github.com/golang/protobuf/proto"
+
+	"google.golang.org/grpc/codes"
+)
+
+func CreateSignalModel(signalID *core.SignalIdentifier, signalType *core.LiteralType, signalValue *core.Literal) (models.Signal, error) {
+	signalModel := models.Signal{}
+	if signalID != nil {
+		signalKey := &signalModel.SignalKey
+		if signalID.ExecutionId != nil {
+			executionKey := &signalKey.ExecutionKey
+			if signalID.ExecutionId.Project != "" {
```

Comment:
nit, I think the preferred check is `len(signalId.ExecutionId.Project) > 0` from Russ Cox's [reply](https://groups.google.com/g/golang-nuts/c/7Ks1iq2s7FA)

Reply:
TIL thanks!